awx/main/models/workflow.py

```diff
@@ -441,9 +441,18 @@ class Meta:
         blank=True,
         default=False,
     )
-    notification_templates_approvals = models.ManyToManyField("NotificationTemplate", blank=True, related_name='%(class)s_notification_templates_for_approvals')
-
-    admin_role = ImplicitRoleField(parent_role=['singleton:' + ROLE_SINGLETON_SYSTEM_ADMINISTRATOR, 'organization.workflow_admin_role'])
+    ask_labels_on_launch = AskForField(
+        blank=True,
+        default=False,
+    )
```

Comment:
What's the history of the request for `ask_labels_on_launch` here? I don't see this field applied to job templates, either here or previously.

Workflows should be essentially mirroring the `ask_` fields from job templates. Applying those prompts to launch JTs in the workflow is the point.

I was talking with @john-westcott-iv regarding new prompt-able fields he's adding, and we would prefer to not add it in multiple places. Instead of adding a new `ask_` field to workflows, I'd advocate that we should _start_ with creating an abstract parent class to house all of those fields, to enforce standardization between the models that define prompts (WFJT, JT), we have a model `LaunchConfigBase` to be the abstract parent of all models that _contain_ prompts, and that is already used here.

Reply:
Looking back over this, we already have a base model that is used strictly by JTs and WFJTs, which is `SurveyJobTemplateMixin`. This makes sense, because surveys are a kind of prompt.

I'd advocate prompting all the `ask_` fields up the inheritance chain from JT to this model. You may want to rename it as well. This would be the abstract base for any "askable" type of model. Then the form to fill out prompts should essentially be the same in all cases, but practically, this means JTs and WFJTs.  Let me know if you see any reason this should not hold true.

Reply:
@john-westcott-iv and myself will merge our JT and WFJT template changes into a common branch and work on the refactor there (https://github.com/ansible/awx/tree/feature-prompt-on-launch-on-templates) as well as work out the failing tests and generate a monolithic migration there.